ruleguard/typematch/typematch.go

```diff
@@ -254,6 +270,31 @@ func parseExpr(ctx *Context, e ast.Expr) *pattern {
 			subs:  append(params, results...),
 		}
 
+	case *ast.StructType:
+		hasSeq := false
+		members := make([]*pattern, 0, len(e.Fields.List))
+		for _, field := range e.Fields.List {
+			p := parseExpr(ctx, field.Type)
+			if p == nil {
+				return nil
+			}
+			if len(field.Names) != 0 {
```

Comment:
probably can be moved above, to prevent parsing when we don't have `field.Names`